File: app/src/main/java/com/xiaomai/geek/ui/module/ReportActivity.java
package com.xiaomai.geek.ui.module;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.Toolbar;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.EditText;

import com.xiaomai.geek.GeekApplication;
import com.xiaomai.geek.R;
import com.xiaomai.geek.di.IComponent;
import com.xiaomai.geek.di.component.DaggerGitHubComponent;
import com.xiaomai.geek.di.component.GitHubComponent;
import com.xiaomai.geek.di.module.ActivityModule;
import com.xiaomai.geek.di.module.GitHubModule;
import com.xiaomai.geek.presenter.github.ReportPresenter;
import com.xiaomai.geek.ui.base.BaseActivity;
import com.xiaomai.geek.view.IReportView;

import javax.inject.Inject;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;

/**
 * Created by XiaoMai on 2017/6/7.
 */

public class ReportActivity extends BaseActivity implements IReportView, IComponent<GitHubComponent> {

    @Inject
    ReportPresenter mPresenter;
    @BindView(R.id.tool_bar)
    Toolbar toolBar;
    @BindView(R.id.et_title)
    EditText etTitle;
    @BindView(R.id.et_content)
    EditText etContent;
    @BindView(R.id.bt_ok)
    Button btOk;

    public static void launch(Context context) {
        context.startActivity(new Intent(context, ReportActivity.class));
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        getComponent().inject(this);
        setContentView(R.layout.activity_report);
        ButterKnife.bind(this);
        initViews();
        mPresenter.attachView(this);
    }

    private void initViews() {
        toolBar.setTitle("意见反馈");
        setSupportActionBar(toolBar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public GitHubComponent getComponent() {
        return DaggerGitHubComponent.builder()
                .applicationComponent(GeekApplication.get(this).getComponent())
                .activityModule(new ActivityModule(this))
                .gitHubModule(new GitHubModule())
                .build();
    }

    @Override
    public void showLoading() {

    }

    @Override
    public void dismissLoading() {

    }

    @Override
    public void reportResult(boolean result) {

    }

    @Override
    public void error(Throwable e) {

    }

    @OnClick(R.id.bt_ok)
    public void onViewClicked() {
        postReport();
    }

    private void postReport() {
        mPresenter.report(this);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        mPresenter.detachView();
    }
}

File: app/src/main/java/com/xiaomai/geek/ui/module/ReportActivity.kt
package com.xiaomai.geek.ui.module

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.MenuItem
import android.widget.Button
import android.widget.EditText
import com.xiaomai.geek.R
import com.xiaomai.geek.ui.base.BaseActivity

class ReportActivity : BaseActivity() {

    var toolBar: Toolbar? = null
    var etTitle: EditText? = null
    var etContent: EditText? = null
    var etContact: EditText? = null
    var btOk: Button? = null

    companion object {
        fun launch(context: Context): Unit {
            context.startActivity(Intent(context, ReportActivity::class.java))
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_report)
        initViews()
    }

    private fun initViews() {
        toolBar = findViewById(R.id.tool_bar) as Toolbar
        etTitle = findViewById(R.id.et_title) as EditText
        etContent = findViewById(R.id.et_content) as EditText
        etContact = findViewById(R.id.et_contact) as EditText
        btOk = findViewById(R.id.bt_ok) as Button

        toolBar?.title = "意见反馈"
        setSupportActionBar(toolBar)
        supportActionBar?.setDisplayHomeAsUpEnabled(true)

        btOk?.setOnClickListener { v ->
            postReport()
        }
    }

    private fun postReport() {

    }

    override fun onOptionsItemSelected(item: MenuItem?): Boolean {
        when (item?.itemId) {
            android.R.id.home -> {
                finish()
                return true
            }
        }
        return super.onOptionsItemSelected(item)
    }
}
